### ISSUE TITLE
Feature Request: Selective Sync

### ISSUE THREAD

**Issue by alberto2000** [nylas/nylas-mail#3402]
This has been raised in multiple issues already - Nylas currently seems to sync all email to the local disc, which eventually leads to tons of GB of data oozing around. While being efficient and comfortable to search or browse through locally cached emails the mass of non-relevant data in comparison to the emails users really need may vary very strongly.

It should be considered to plan some kind of "Selective Sync" - either an option to select the folders users want to be synced or an option to limit the data that will be used for cache.

See Newton Email for example https://newtonhq.com afaik they only locally cache the inbox folder, all other email is fetched on request. It works great on my Macbook and I love the sleekness and sustainable approach it takes on users hard disk space.

**Comment by psychonaut** [comment]
This feature is very needed, i've connected my three accounts in Nylas Mail and after 3 days it still syncing. That wouldn't be a problem if Inbox has priority in syncing but it doesn't.

**Comment by milesimarco** [comment]
Same problem with high CPU and disk usage.
With years of email history and multiple accounts, Selective Sync would be very appreciated.

**Comment by michalhudecek** [comment]
Yes, especially for spam folder. CPU is skyrocketing and all that is being downloaded are spam messages.

**Comment by itzsaga** [comment]
Came up again in #3501

**Comment by elhenro** [comment]
Also having sync issues.. I wouldnt actually want to sync the complete mail cache..

**Comment by itzsaga** [comment]
Also in #3516

**Comment by itzsaga** [comment]
I selective sync for full download and a headers download for fast local search would be a nice implementation IMO. Can't remember which other client does this but it was super fast AND light on my system. I dumped it due to lack of multiple accounts.

**Comment by olympum** [comment]
Voting that this a critical feature for me AFAIK - I love the interface, the simplicity, ... but just can't accept 3 days with the CPU running on and on syncing and indexing.

**Comment by moxx** [comment]
+1 absolutely agree, without this feature the mail client is still unusable if you have mail achievement-duty of 10years like in germany and austria!
